### PR TITLE
Enable logic for GPU auto-detection in cudfjni

### PR DESCRIPTION
**Pull request by gerashegalov**
Allow overriding `GPU_ARCHS` with an empty string in cudfjni to enable automatic detection
```bash
mvn clean install -DARROW_STATIC_LIB=ON -DBoost_USE_STATIC_LIBS=ON -DGPU_ARCHS=
...
     [exec] -- CUDA_VERSION: 11.0
     [exec] Auto detection of gpu-archs: 75
     [exec] GPU_ARCHS = 75
```

Allow `--h[elp]`  switch to `$CUDF_HOME/build.sh`

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -37,7 +37,7 @@ HELP="$0 [clean] [libcudf] [cudf] [dask_cudf] [benchmarks] [tests] [libcudf_kafk
    --disable_nvtx       - disable inserting NVTX profiling ranges
```
Comment (jlowe) [inline build.sh:37]: Need to add 2021 copyrights on this file

File: java/pom.xml [hunk 2]
```diff
@@ -148,6 +148,7 @@
         <PER_THREAD_DEFAULT_STREAM>OFF</PER_THREAD_DEFAULT_STREAM>
```
Comment (jlowe) [inline java/pom.xml:148]: Add 2021 copyright

**Review by jlowe — APPROVED** [review]
Needs 2021 copyrights on the files but otherwise lgtm.

File: build.sh (hunk 1 above)
Comment (gerashegalov) [reply build.sh:37]: ack, would be great to add stuff like this to a pre-commit hook

**Review by kkraus14 — APPROVED** [review]
Approving build.sh changes

**Pull request merged**